src/impulsoetl/utilitarios/bd.py

```diff
@@ -3,28 +3,245 @@
+        sob demanda quando requisitadas pelo método `__getitem__()`
+        (equivalente a obter a representação de uma tabela do dicionário
+        chamando `dicionario["nome_do_schema.nome_da_tabela"]`).
+
+        Argumentos:
+            metadata_obj: instância da classe [`sqlalchemy.schema.MetaData`][]
+                da biblioteca SQLAlchemy,
+            **kwargs: Parâmetros adicionais a serem passados para o método
+                [`reflect()`][] do objeto de metadados ao se tentar obter uma
+                tabela ainda não espelhada no banco de dados.
+
+        [`sqlalchemy.schema.MetaData`]: https://docs.sqlalchemy.org/en/14/core/metadata.html#sqlalchemy.schema.MetaData
+        [`reflect()`][]: https://docs.sqlalchemy.org/en/14/core/metadata.html#sqlalchemy.schema.MetaData.reflect
+        """
+        self.meta = metadata_obj
+        self.kwargs = kwargs
+
+    def __getitem__(self, chave: str) -> Table:
+        try:
+            return self.meta.tables[chave]
+        except (InvalidRequestError, KeyError):
+            schema = None
+            try:
+                schema, tabela_nome = chave.split(".", maxsplit=1)
+            except ValueError:
+                tabela_nome = chave
+            logger.debug("Espelhando tabela `{}`...", chave)
+            self.meta.reflect(schema=schema, only=[tabela_nome], **self.kwargs)
+            logger.debug("OK.")
+            return self.meta.tables[chave]
+
+    def __setitem__(self, chave: str, valor: Table) -> None:
+        self.meta.tables[chave] = valor
+
+    def __repr__(self) -> str:
+        return self.meta.tables.__repr__()
+
+    def update(self, *args, **kwargs) -> None:
+        for chave, valor in dict(*args, **kwargs).items():
+            self[chave] = valor
+
+
+def postgresql_copiar_dados(
+    tabela_dados: SQLTable,
+    conexao: Connection | Engine,
+    colunas: Iterable[str],
+    dados_iterador: Iterable,
+) -> None:
+    """Inserir dados em uma tabela usando o comando COPY do PostgreSQL.
+
+    Esta função deve ser passada como valor para o argumento `method` do
+    método [`pandas.DataFrame.to_sql()`][].
+
+    Argumentos:
+        tabela_dados: objeto [`pandas.io.sql.SQLTable`][] com a representação
+            da tabela SQL a ser inserida.
+        conexao: objeto [`sqlalchemy.engine.Engine`][] ou
+            [`sqlalchemy.engine.Connection`][] contendo a conexão de alto nível
+            com o banco de dados gerenciada pelo SQLAlchemy.
+        colunas: lista com os nomes de colunas a serem inseridas.
+        dados_iterador: Iterável com os dados a serem inseridos.
+
+    Exceções:
+        Levanta uma subclasse da exceção [`psycopg2.Error`][] caso algum erro
+        seja retornado pelo backend.
+
+    Veja também:
+        - [Documentação][io-sql-method] do Pandas sobre a implementação de
+        funções personalizadas de inserção.
+        - [Artigo][insert-a-pandas-dataframe-into-postgres] com comparação da
```

Comment:
Eu li o artigo e não achei motivação para a mudança para "psycopg2 using copy_from and csv
" dado que "pandas.DataFrame.to_sql with method=callable" perfoma tão bem quanto e não precisa converter o dataframe. Entendo que agora já está assim e não precisamos mudar só não entendi a principal motivação.

Edit: Esquece, agora que vi o código ali embaixo.